src/host/outputStream.cpp

```diff
@@ -286,7 +286,17 @@ void ConhostInternalGetSet::ShowWindow(bool showOrHide)
 {
     const auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
     const auto hwnd = gci.IsInVtIoMode() ? ServiceLocator::LocatePseudoWindow() : ServiceLocator::LocateConsoleWindow()->GetWindowHandle();
+
+    // GH#13301 - When we send this ShowWindow message, if we send it to the
+    // conhost HWND, it's going to need to get processed by the window message
+    // thread before returning. We (the VT adapter) are handling this message
+    // under lock. However, the first thing the conhost message thread does is
+    // lock the console. That'll deadlock us. So unlock here, first, to let the
+    // message thread deal with this message, then re-lock so later on this
+    // thread can unlock again safely.
+    gci.UnlockConsole();
     ::ShowWindow(hwnd, showOrHide ? SW_SHOWNOACTIVATE : SW_MINIMIZE);
+    gci.LockConsole();

```

Comment:
We did something similar for MIDI recently. So I think this is fine.
Do we need to wait for `ShowWindow` to complete? If not `ShowWindowAsync` can be used instead.
BTW I'm surprised this doesn't use `SW_RESTORE`.

Reply:
![what-000](https://user-images.githubusercontent.com/18356694/174164127-e296084e-341e-486f-b7e2-37163d5a69aa.gif)

Reply:
I literally till this moment did not know there was a `ShowWindowAsync`. That needs to be SO MUCH HIGHER in that doc.